cmd/server/helper_keys.go

```diff
@@ -18,7 +18,7 @@ func createRS256KeysIfNotExist(c *config.Config, set, kid, use string) {
 	if _, err := ctx.KeyManager.GetKey(set, kid); errors.Cause(err) == pkg.ErrNotFound {
 		logrus.Infof("Key pair for signing %s is missing. Creating new one.", set)
 
-		keys, err := generator.Generate("")
+		keys, err := generator.Generate(set)
```

Comment:
So the confusion here is stemming from which keys are returned in the `well-known` response. I am assuming that all of the public keys are returned, hence each set would have a `kid` of `public` and then you put all of those public keys in a single set and return them. Leading to 3 keys with the `kid` of `public`. 

Whats interesting as I read through the OpenID spec http://openid.net/specs/openid-connect-discovery-1_0.html#ProviderMetadata again, they don't specify which key types should be returned. Looking at Okta http://developer.okta.com/docs/api/resources/oauth2.html#get-authorization-server-keys they are returning just one key type, but with an added `status` parameter.

I was seeing 3 keys from google and ms thinking it was `consent, challenge, and idtoken`, but it may be `past, present, future` of the same key.